src/single-spa-react.js

```diff
@@ -74,6 +74,11 @@ function bootstrap(opts, props) {
 
 function mount(opts, props) {
   return new Promise((resolve, reject) => {
+
+    if (!opts.rootComponent.prototype.componentDidCatch && !opts.suppressComponentDidCatchWarning && atLeastReact16(opts.React)) {
```

Comment:
Will this work with stateless function components? I think it probably will, maybe we could we add a test for this to double check.